### PR TITLE
[test] [example] Add a test for print_offset example

### PR DESCRIPTION
**Pull request by LittleFall**
Related issue = #3580

### CODE REVIEW

**Comment by LittleFall** [comment]
test failed on `Build and Test Windows`:
```
Run actions/checkout@v2
Syncing repository: taichi-dev/taichi
Getting Git version info
Deleting the contents of 'C:\actions-runner\_work\taichi\taichi'
Error: EPERM: operation not permitted, unlink 'C:\actions-runner\_work\taichi\taichi\venv'
```

File: tests/python/examples/algorithm/test_print_offset.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import argparse
+
+from taichi.lang import impl
+
+import taichi as ti
+
+FRAMES = 100
+
+
+def test_print_diff():
```
Comment (strongoier) [inline tests/python/examples/algorithm/test_print_offset.py:10]: Rename to `print_offset`?

Comment (LittleFall) [reply tests/python/examples/algorithm/test_print_offset.py]: thanks

File: tests/python/examples/algorithm/test_print_offset.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+from taichi.lang import impl
+
+import taichi as ti
+
+FRAMES = 100
+
+
+def test_print_offset():
+    from taichi.examples.algorithm.print_offset import a, fill, m, n
```
Comment (strongoier) [inline tests/python/examples/algorithm/test_print_offset.py:11]:
```suggestion
    from taichi.examples.algorithm.print_offset import fill
```

File: tests/python/examples/algorithm/test_print_offset.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def video_print_offset(result_dir):
+    from taichi.examples.algorithm.print_offset import a, fill, m, n
+    video_manager = ti.VideoManager(output_dir=result_dir,
+                                    framerate=24,
+                                    automatic_build=False)
+
+    fill()
+    print(a.to_numpy())
```
Comment (strongoier) [inline tests/python/examples/algorithm/test_print_offset.py:22]:
```suggestion
```

File: tests/python/examples/algorithm/test_print_offset.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    impl.get_runtime().prog.visualize_layout('layout.pdf')
```
Comment (strongoier) [inline tests/python/examples/algorithm/test_print_offset.py:24]:
```suggestion
```

**Review by strongoier — APPROVED** [review]
Thanks for your contribution! LGTM besides some nits :-)

**Comment by LittleFall** [comment]
test cancelled on `Build and Test Windows`, how can I rerun it?
```
  C:\actions-runner\_work\taichi\taichi\tests\python\test_function.py:142: UserWarning: Calling non-taichi function "identity". Scope inside the function is not processed by the Taichi AST transformer. The function may not work as expected. Proceed with caution! Maybe you can consider turning it into a @ti.func?
    inc(identity(3))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! KeyboardInterrupt !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
C:\actions-runner\_work\_tool\Python\3.7.9\x64\lib\threading.py:300: KeyboardInterrupt
(to show a full traceback on KeyboardInterrupt use --full-trace)
========== 539 passed, 2 skipped, 232 warnings in 394.00s (0:06:33) ===========
Error: The operation was canceled.
```

**Pull request merged**